draft-ietf-tls-dtls-connection-id.md

```diff
@@ -159,161 +157,79 @@ for example by having the length in question be a compile-time constant.
 
-~~~~
-  struct {
-     opaque content[DTLSPlaintext.length];
-     ContentType type;
-     uint8 zeros[length_of_padding];
-  } DTLSInnerPlaintext;
-
-  struct {
-     ContentType opaque_type = 23; /* application_data */
-     ProtocolVersion legacy_record_version = {254,253); // DTLSv1.2
-     uint16 epoch;                         // DTLS-related field
-     uint48 sequence_number;               // DTLS-related field
-     opaque cid[cid_length];               // New field
-     uint16 length;
-     opaque encrypted_record[length];
-  } DTLSCiphertext;
-~~~~
-{: #dtls-record13 title="DTLS 1.3 Record Format with Connection ID"}
-
-Besides the "cid" field, all other fields are defined in the DTLS 1.2 and
-DTLS 1.3 specifications.
-
 Note that for both record formats, it is not possible to parse the
-records without knowing if the connection ID is in use and how long
-it is.
+records without knowing how long the Connection ID is.
 
-# Examples
+In order to allow a receiver to determine whether a record has CID or not,
+connections which have negotiated this extension use new record types for all
+protected records. {{new-cid-content-types}} shows the record types to use:
 
-Below is an example exchange for DTLS 1.3 using a single
-connection id in each direction.
+| New ContentType | Value |
+|--------------|-----------|
+| alert_with_cid | 25 |
+| handshake_with_cid | 26 |
+| application_data_with_cid | 27 |
+| heartbeat_with_cid | 28 |
+{: #new-cid-content-types}
 
-~~~~
-Client                                             Server
-------                                             ------
+# CID authentication
 
-ClientHello
-(connection_id=5)
-                            -------->
-
-
-                            <--------       HelloRetryRequest
-                                                     (cookie)
+The CID is authenticated.  The MAC of a DTLS record with CID is generated as:
 
-ClientHello                 -------->
-(connection_id=5)
-  +cookie
-
-                            <--------             ServerHello
-                                          (connection_id=100)
-                                          EncryptedExtensions
-                                                      (cid=5)
-                                                  Certificate
-                                                      (cid=5)
-                                            CertificateVerify
-                                                      (cid=5)
-                                                     Finished
-                                                      (cid=5)
+~~~~
+      MAC(MAC_write_key, DTLSCompressed.epoch +
+                            DTLSCompressed.sequence_number +
+                            DTLSCompressed.type +
+                            DTLSCompressed.version +
+                            DTLSCompressed.cid +      // New input
+                            DTLSCompressed.length +
```

Comment:
Do we need to modify the key derivation function to include the CID? Why do you think we need it?

Reply:
(This the MAC computation, not the key derivation. Though, this is probably not needed -- see my reply above.)